api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationOAuth.java

```diff
@@ -141,6 +157,28 @@ public void setUserNameClaim(String userNameClaim) {
         this.userNameClaim = userNameClaim;
     }
 
+    @Description("The fallback username claim to be used for user id if the claim specified by `userNameClaim` is not present. " +
+            "Useful when `client_credentials` authentication only results in client id being provided in another claim. " +
+            "It only takes effect if `userNameClaim` is set.")
+    public String getFallbackUserNameClaim() {
+        return fallbackUserNameClaim;
+    }
+
+    public void setFallbackUserNameClaim(String fallbackUserNameClaim) {
+        this.fallbackUserNameClaim = fallbackUserNameClaim;
+    }
+
+    @Description("The prefix to use with the value of `fallbackUserNameClaim` to construct the user id. " +
+            "It only takes effect if `fallbackUserNameClaim` is set, and the value is present for the claim. " +
+            "Useful to be able to map usernames and client ids into the same user id space, preventing name collisions.")
+    public String getFallbackUserNamePrefix() {
+        return fallbackUserNamePrefix;
+    }
+
+    public void setFallbackUserNamePrefix(String fallbackUserNamePrefix) {
+        this.fallbackUserNamePrefix = fallbackUserNamePrefix;
+    }
+
     @Description("Configure whether the access token type check should be performed or not. This should be set to `false` " +
```

Comment:
St to 'false' in any situation?

Reply:
If this was the whole sentece the idea would be that by default it is `true`, and if you're setting it, you probably want to set it to `false`. But the sentence continues with 'if the authorization server does not include 'typ'' - so that's when it should be set to 'false'.

Reply:
@PaulRMellor Thanks